src/wallet/test/pos_validations_tests.cpp

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2020 The PIVX developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or https://www.opensource.org/licenses/mit-license.php.
+
+#include "wallet/test/pos_test_fixture.h"
+
+#include "blockassembler.h"
+#include "util/blockstatecatcher.h"
+#include "blocksignature.h"
+#include "consensus/merkle.h"
+#include "primitives/block.h"
+#include "script/sign.h"
+#include "test/util/blocksutil.h"
+#include "wallet/wallet.h"
+
+#include <boost/test/unit_test.hpp>
+
+BOOST_AUTO_TEST_SUITE(pos_validations_tests)
+
+void reSignTx(CMutableTransaction& mtx,
+              const std::vector<CTxOut>& txPrevOutputs,
+              CWallet* wallet)
+{
+    CTransaction txNewConst(mtx);
+    for (int index=0; index < txPrevOutputs.size(); index++) {
```

Comment:
just to remove `-Wsign-compare` warnings
```suggestion
    for (int index=0; index < (int) txPrevOutputs.size(); index++) {
```